### PR TITLE
qt: add "kde" to qt.platformTheme

### PR DESCRIPTION
**Pull request by thiagokokada**
### Description
This allow you to configure Qt integration using KDE instead of `qgnomeplatform` or `qtstyleplugins`. Useful if your theme supports both GTK and KDE, for example Nordic.

To use this properly you will need to do some manual configuration for now. You can set the theme settings using `~/.config/kdeglobals`.

Example:

```nix
{ ... }:
{
  qt = {
    enable = true;
    platformTheme = "kde";
  };

  xdg = {
    configFile.kdeglobals.text = lib.generators.toINI { } {
      General = {
        ColorScheme = "nordicbluish";
        Name = "nordic-bluish";
        shadeSortColumn = true;
      };
      Icons = {
        Theme = "Nordic-bluish";
      };
      KDE = {
        LookAndFeelPackage = "Nordic-bluish";
        contrast = 4;
      };
    };

    dataFile = {
      # For General.ColorScheme
      color-schemes = {
        source = "${pkgs.nordic}/share/color-schemes";
        recursive = true;
      };
      # For KDE.LookAndFeelPackage
      plasma = {
        source = "${pkgs.nordic}/share/plasma";
        recursive = true;
      };
    };
  };
}
```

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by thiagokokada** [comment]
CC @berbiche @rycee.

File: modules/misc/qt.nix [hunk 1]
```diff
@@ -133,15 +137,22 @@ in {
...
     # Necessary because home.sessionVariables doesn't support mkIf
     home.sessionVariables = filterAttrs (n: v: v != null) {
-      QT_QPA_PLATFORMTHEME =
-        if cfg.platformTheme == "gnome" then "gnome" else "gtk2";
+      QT_QPA_PLATFORMTHEME = if cfg.platformTheme == "gnome" then
+        "gnome"
+      else if cfg.platformTheme == "kde" then
+        "kde"
+      else
+        "gtk2";
```
Comment (ncfavier) [inline modules/misc/qt.nix:145]:
```suggestion
      QT_QPA_PLATFORMTHEME = if cfg.platformTheme == "gtk" then
        "gtk2"
      else cfg.platformTheme;
```

Comment (thiagokokada) [reply modules/misc/qt.nix]: Done.

**Review by ncfavier — APPROVED** [review]

File: tests/modules/misc/qt/qt-platform-theme-kde.nix [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{ ... }:
+
+{
+  config = {
+    qt = {
+      enable = true;
+      platformTheme = "kde";
+    };
+
```
[the remark this answers is not in the document]
Comment (ncfavier) [reply tests/modules/misc/qt/qt-platform-theme-kde.nix:9]: Yeah this test is not very useful

[the remark this answers is not in the document]
Comment (thiagokokada) [reply tests/modules/misc/qt/qt-platform-theme-kde.nix]: It did help me find an issue during development. But yes, it is not that useful.

Removed.

**Comment by thiagokokada** [comment]
@ncfavier Can we get this merged?

**Pull request merged**